Coding: Develop a timeline for programs page (#73)

iamkryptonite opened the pull request:
### Description

Developed a timeline for the events page. 

Fixes #73

### Type of Change:
**Delete irrelevant options.**

- Code
- User Interface

**Code/Quality Assurance Only**
- Bugfix (a non-breaking change that fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)



### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.
![AnitaB_org_open_source](https://user-images.githubusercontent.com/43510609/96408175-91c77500-1200-11eb-9319-d2d9df3809b7.png)


![ezgif com-video-to-gif](https://user-images.githubusercontent.com/43510609/96409065-53cb5080-1202-11eb-9b2a-648061f0cdea.gif)


**gh-pages Link:** https://iamkryptonite.github.io/anitab-org.github.io/




### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented on my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,99 @@
+import $ from 'jquery';
+import {ArrowLeftOutlined ,ArrowRightOutlined} from '@ant-design/icons'
+import dayjs from 'dayjs'
+const events=[
```
this should be kept in content js so we can edit

iamkryptonite replied:
okay

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,99 @@
+        scroll.scrollLeft(p+=35);
+    }
+  return (
+      <>
```
what this line do?

iamkryptonite replied:
It handles the onClick events . When the user clicks the past or future arrows, it scrolls the timeline accordingly. That long className targets the scrollview.

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,99 @@
+                            {
+                                months.map((m)=>(
+                                    <>
+                                        <View style={{left:(m[0]*30)*5+1+'%',marginBottom:'15px',top:'-44px'}}><Text>1st {m[1]}</Text></View>
```
styles should be hold using StyleSheet imported from react native

iamkryptonite replied:
I have made some changes although there are still some inline styles which cannot be changed .

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,99 @@
+                    <View style={{width:'90%'}}>
+                        {
+                            events.map((item)=>(
+                                <View style={{marginBottom:'3vw',flexDirection:'column'}} key={item.event}>
```
3vw will break native app ones we move it there.

annabauza commented:
this looks great we just need to tweak the code a bit

iamkryptonite commented:
Hey @annabauza I have made the necessary changes in the code ,could you please review it once

nandini45 commented:
@Bucky25 can you review this one?

Bucky25 commented on src/Components/Programs/Timeline.js:
```diff
@@ -62,13 +55,13 @@ function Timeline() {
                         {
                             events.map((item)=>(
-                                <View style={{marginBottom:'3vw',flexDirection:'column'}} key={item.event}>
+                                <View style={{marginBottom:'47px',flexDirection:'column'}} key={item.event}>
```
please make it  a power of 2

Bucky25 commented on src/Components/Programs/styles.js:
```diff
@@ -58,6 +57,35 @@ export const Stroke = styled(View,{
+    width:'80%'
+})
+export const Months = styled(View,{
+    marginBottom:'15px',
```
please make it power of 2

iamkryptonite replied:
Okay

Bucky25 commented on src/Components/Programs/styles.js:
```diff
@@ -58,6 +57,35 @@ export const Stroke = styled(View,{
+})
+export const ArrowNavigation = styled(View,{
+    flexDirection:'row',
+    marginTop:'10px',
+    marginBottom:'10px',
```
please make it power of 2

Bucky25 replied:
and please add a link to your deployed gh pages

iamkryptonite replied:
gh-pages link: https://iamkryptonite.github.io/anitab-org.github.io/

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    paddingVertical:4,
+    paddingHorizontal: 8,
+    marginBottom: 4,
+    borderRadius: '50px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    marginBottom: 4,
+    borderRadius: '50px',
+    textAlign: 'center',
+    zIndex:'1px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    zIndex:'1px',
+    position:'relative',
+    backgroundColor:'#fff',
+    width:'200px'
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    width:'200px'
+});
+export const Line = styled(View,{
+    width:'200vw',
```
we don't use vw because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    borderBottomWidth:10,
+    marginBottom:4,
+    position:'relative',
+    top: '18px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    fontSize:20,
+    width:45,
+    height:45,
+    borderRadius: '50px',
```
we don't use px because this will break moving to modules to native app.

iamkryptonite replied:
Without the 'px' border-radius doesn't seem to work. The same problem occurs for z-index also.

iamkryptonite replied:
@annabauza

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    width:'2px',
+    height: '60px',
+    position:'absolute',
+    top: '28px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+});
+export const Marker = styled(View,{
+    backgroundColor:'#000',
+    width:'2px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+export const Marker = styled(View,{
+    backgroundColor:'#000',
+    width:'2px',
+    height: '60px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+});
+export const Fade = styled(View,{
+    backgroundColor:'#fff',
+    height:'90px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+})
+export const Stroke = styled(View,{
+    borderBottomColor:'#000',
+    borderBottomWidth:'2px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+export const Stroke = styled(View,{
+    borderBottomColor:'#000',
+    borderBottomWidth:'2px',
+    width:'50vw',
```
we don't use vw because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    borderBottomColor:'#000',
+    borderBottomWidth:'2px',
+    width:'50vw',
+    top:'-45px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+})
+export const ArrowNavigation = styled(View,{
+    flexDirection:'row',
+    marginTop:'8px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+export const ArrowNavigation = styled(View,{
+    flexDirection:'row',
+    marginTop:'8px',
+    marginBottom:'8px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    justifyContent:'space-between'
+})
+export const ArrowText = styled(Text,{
+    fontSize:'16px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+})
+export const ArrowText = styled(Text,{
+    fontSize:'16px',
+    marginLeft:'8px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+export const ArrowText = styled(Text,{
+    fontSize:'16px',
+    marginLeft:'8px',
+    marginRight:'8px'
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    width:'80%',
+    flexWrap:'wrap',
+    flexDirection:'column',
+    marginBottom:'64px'
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+    width:'80%'
+})
+export const Months = styled(View,{
+    marginBottom:'16px',
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/styles.js:
```diff
@@ -0,0 +1,85 @@
+})
+export const Months = styled(View,{
+    marginBottom:'16px',
+    top:'-35px'
```
we don't use px because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,92 @@
+                    <Container>
+                        {
+                            events.map((item)=>(
+                                <View style={{marginBottom:'32px',flexDirection:'column'}} key={item.event}>
```
this style can be moved to styles, right?

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,92 @@
+                {
+                    events.map((item)=>(
+                        <View key={item.event}>
+                            <Event style={{color:item.color,borderColor:item.color,flex:1,left:item.date[0][1]+'vw'}}>{item.event}</Event>
```
we don't use px or vw, because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,92 @@
+                <View style={{flexDirection:'row'}}>
+                    <ArrowText>FUTURE</ArrowText>
+                    <TouchableWithoutFeedback onPress={scrollRight}>
+                        <ArrowRightOutlined style={{color:'#42aaf5',paddingTop:'4px'}}/>
```
we don't use px or vw, because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,92 @@
+            <ArrowNavigation>
+                <View style={{flexDirection:'row'}}>
+                    <TouchableWithoutFeedback onPress={scrollLeft}>
+                        <ArrowLeftOutlined style={{color:'#42aaf5',paddingTop:'4px'}}/>
```
we don't use px or vw, because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,92 @@
+                        {
+                            events.map((item)=>(
+                                <View style={{marginBottom:'32px',flexDirection:'column'}} key={item.event}>
+                                    <Line style={{left:(item.date[0][1]-1)*5+'%', width:30*12*5+'vw', borderBottomColor:item.color}}></Line>
```
we don't use px or vw, because this will break moving to modules to native app.

annabauza commented on src/Components/Programs/Timeline.js:
```diff
@@ -0,0 +1,92 @@
+    var curr_month = dayjs().month(); //current month
+    //scrolls the timeline to the current date and month
+    useEffect(() => {
+        var scroll = $('.css-view-1dbjc4n');
```
is `css-view-1dbjc4n` obfuscated reference?

annabauza reviewed (changes requested):
Generally very good code. I have left you mostly notes related to css. Thanks.

annabauza commented:
@iamkryptonite can you please also resolve your conflicts? Thanks

tichnas reviewed (changes requested):
Hi, it looks like the PR will work only for the web version. We should not use things like classes and jQuery in React Native. Please do these changes.
Thanks

iamkryptonite commented:
> 
> 
> Hi, it looks like the PR will work only for the web version. We should not use things like classes and jQuery in React Native. Please do these changes.
> Thanks

Hi,I have made the changes. Please review once.

tichnas commented:
@iamkryptonite I think you're still using jQuery and pixels which don't go well with React Native. Can you please change them also?
Thanks for giving time to do these changes :)

iamkryptonite commented:
> 
> 
> @iamkryptonite I think you're still using jQuery and pixels which don't go well with React Native. Can you please change them also?
> Thanks for giving time to do these changes :)

 Hi @tichnas, I have removed jquery and used useRef instead. And I have also changed all the places where pixels were used.

annabauza reviewed (approved):
💯 This is great thank you!

annabauza commented:
@iamkryptonite please call yarn in command line and commit yarn.lock because travis is failing.

tichnas reviewed (changes requested):
Please commit the changes to `yarn.lock` (and don't delete the file)
Also, remove your homepage link from `package.json` and can you please tell what's `@babel/plugin-transform-react-jsx` used for?

Thanks

iamkryptonite commented:
> 
> 
> Please commit the changes to `yarn.lock` (and don't delete the file)
> Also, remove your homepage link from `package.json` and can you please tell what's `@babel/plugin-transform-react-jsx` used for?
> 
> Thanks

Hey, I have made the requested changes and also removed the babel package as it was not required.

annabauza reviewed (changes requested):
please resolve conflict and merge with existing events tab. Thanks!

annabauza commented on package.json:
```diff
@@ -1,11 +1,13 @@
 {
-  "homepage": "./",
+  "homepage": "/",
```
Why this change?

iamkryptonite replied:
I am not really sure, I think this was a mistake so have changed it.

annabauza reviewed (approved):
Generally OK for first iteration just a quick question does homepage change would not break github static pages?

Vishal-raj-1 reviewed (approved):
Great Work @iamkryptonite !! Could you please squash your commits

annabauza commented:
@Vishal-raj-1 we are squashing comments on merge, no need to do that.

iamkryptonite commented:
Hey @annabauza there is one small change that I need to make in my code.

annabauza commented:
Rise the issue and ask to be assigned

iamkryptonite commented:
> Rise the issue and ask to be assigned

Hey @annabauza I have raised the [issue](https://github.com/anitab-org/anitab-org.github.io/issues/246). Could you pls approve it and add relevant labels.
Thanks.
